### PR TITLE
Makes local host admin by default

### PR DESCRIPTION
**Pull request by ghost**
Clients joining with loopback ip (127.0.0.1) or something of the sort are automatically given the Host rank upon joining. Has config option so it can be easily disabled or enabled.

This took me way longer to figure out than it should have.

🆑 

* rscadd: By default, clients joining with a local / loopback IP will receive the Host rank, this can be disabled via config.

### CODE REVIEW

**Comment by Probe1** [comment]
yes we should definitely give anyone that can spoof their ip +Host

this wont cause issues.

File: code/modules/client/client procs.dm [hunk 1]
```diff
@@ -140,6 +140,11 @@
 		admins += src
 		holder.owner = src
 
+	var/localhost_addresses = list("127.0.0.1","::1")
+	if(!address || (address in localhost_addresses))
```
Comment (MrsTonedOne) [inline code/modules/client/client procs.dm]: While doing the !address bit is required to give people who launch the code in DS standalone host, its the biggest thing you might want to worry about.

There was already once an exploit that allowed somebody make their address and cid in isbanned() be null, there might be another that effects client.version, allowing somebody to spoof.

Change this to `if ((!address && !world.port) || ...` to be more secure.

Comment (Probe1) [reply code/modules/client/client procs.dm]: Thanks MSO I was gonna ask you after @N3X15

**Comment by Aurx** [comment]
Not having to copy/paste something into the config folder isn't worth a security risk.  

Please find a way to disable this on the live serb but enable it on local serbs, like some config shit or something.

**Comment by PJB3005** [comment]
The fact that we have to account for people being able to spoof their IP to local host with BYOND is sad.

**Comment by Aurx** [comment]
Thanks BYOND, we expect nothing but the most basic of things and are constantly disappointed.

**Comment by clusterfack** [comment]
Make a config to disable it and put that config in the secret repo

**Comment by PJB3005** [comment]
Exactly my thoughts.

**Comment by ghost** [comment]
Should be little to no security risk now (this is actually almost the exact same thing tgstation does), but I'd definitely separate this in the secret repo just to be safe.

**Comment by PJB3005** [comment]
Make it a config option so we can disable it secret repo side easily.

**Comment by ghost** [comment]
There you go.

**Review by PJB3005 — APPROVED** [review]

File: config-example/config.txt [hunk 2]
```diff
@@ -28,6 +28,9 @@ JOBS_HAVE_MINIMAL_ACCESS
 ## Unhash this to use recursive explosions, keep it hashed to use circle explosions. Recursive explosions react to walls, airlocks and blast doors, making them look a lot cooler than the boring old circular explosions. They require more CPU and are (as of january 2013) experimental
 #USE_RECURSIVE_EXPLOSIONS
 
+## Automatically give localhost clients +HOST access
```
Comment (N3X15) [inline config-example/config.txt]: Please add a "FUCKING DISABLE THIS ON PRODUCTION SERVERS OR YOU WILL GET REKT" comment

**Comment by ghost** [comment]
merge when?
a month old and nobody wants to fix a tiny secret repo conflict apparently

**Comment by Probe1** [comment]
I'm not gonna be the one to merge what could be a huge security vulnerability for QoL tbh.   I'm not saying no one else can merge it but I sure as hell am not gonna stick my foot in it.

**Comment by ghost** [comment]
but it really isn't much of a security vulnerability with its own config toggle
i don't think you can spoof your ip like this either actually
just disable it in the secret repo config

**Comment by MrsTonedOne** [comment]
This is actually pretty secure in its current form, its more secure than the version /tg/ has (something i might look at changing)

**Pull request merged**